### PR TITLE
C additional properties fix #4289

### LINKED ISSUE #4289: [BUG][C] errors in supporting additionalProperties in C client
#### Bug Report Checklist

- [ X ] Have you provided a full/minimal spec to reproduce the issue?
- [ X ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ X ] What's the version of OpenAPI Generator used?
- [ X ] Have you search for related issues/PRs?
- [ X ] What's the actual output vs expected output?

##### Description

I tried to compile the declaration reported below. The results was:

/mnt/hgfs/shared/clientC/model/service_query_form_dto.c:25:15: error: ‘localKeyValue’ undeclared (first use in this function); did you mean ‘localMapKeyPair’?
         free (localKeyValue->key);
               ^~~~~~~~~~~~~
               localMapKeyPair

##### openapi-generator version

commit 4543c210304c25a551a6a9c50d330dfd6f3ca462 (grafted, HEAD -> master, origin/master, origin/HEAD)
Author: sullis <github@seansullivan.com>
Date:   Sat Oct 26 21:10:55 2019 -0700

##### OpenAPI declaration file content or url

https://gist.github.com/michelealbano/4ecf5b628a5d7eb305a3eea5da16ff4a

##### Command line used for generation

java -jar openapi-generator/modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g c -i cproblems.json -o clientC --additional-properties=swaggerDocketConfig=true

##### Steps to reproduce

java -jar openapi-generator/modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -g c -i cproblems.json -o clientC --additional-properties=swaggerDocketConfig=true
cd clientC
mkdir build
cd build
cmake ..
make

##### Related issues/PRs

N/A

##### Suggest a fix

In model-body.mustache, change "localMapKeyPair" on line 258 into "localKeyValue".

### PR DESCRIPTION
**Pull request by michelealbano**
Solved #4289

### PR checklist

- [ X ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ X ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [ X ] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [ X ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache [hunk 1]
```diff
@@ -526,8 +526,8 @@ fail:
     char* decoded = base64decode({{{baseName}}}->valuestring, strlen({{{baseName}}}->valuestring));
     decoded_str_{{{baseName}}}->data = malloc(strlen(decoded) - 1);
     if (!decoded_str_{{{baseName}}}->data) {
-		goto end;
-	}
+        goto end;
+    }
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache]: @michelealbano thanks for fixing the formatting in the templates.

Another way is to use a code formatter such as https://clang.llvm.org/docs/ClangFormat.html to format the output.

File: samples/client/petstore/c/CMakeLists.txt [hunk 2]
```diff
@@ -52,34 +52,39 @@ set(HDRS
...
+#set(SRCS
+#    unit-tests/manual-PetAPI.c
+#    unit-tests/manual-StoreAPI.c
+#    unit-tests/manual-UserAPI.c
+#    unit-tests/manual-order.c
+#    unit-tests/manual-user.c)
+
+##set header files
+#set(HDRS
+#)
```
Comment (wing328) [inline samples/client/petstore/c/CMakeLists.txt]: @michelealbano seems like these will disable the integration tests for the C Petstore client. I could no longer find the following in the Travis log:
```
-- Build files have been written to: /home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/c
Scanning dependencies of target openapi_petstore
[  4%] Building C object CMakeFiles/openapi_petstore.dir/src/list.c.o
[  8%] Building C object CMakeFiles/openapi_petstore.dir/src/apiKey.c.o
[ 12%] Building C object CMakeFiles/openapi_petstore.dir/src/apiClient.c.o
[ 16%] Building C object CMakeFiles/openapi_petstore.dir/external/cJSON.c.o
[ 20%] Building C object CMakeFiles/openapi_petstore.dir/model/api_response.c.o
[ 25%] Building C object CMakeFiles/openapi_petstore.dir/model/category.c.o
[ 29%] Building C object CMakeFiles/openapi_petstore.dir/model/order.c.o
[ 33%] Building C object CMakeFiles/openapi_petstore.dir/model/pet.c.o
[ 37%] Building C object CMakeFiles/openapi_petstore.dir/model/tag.c.o
[ 41%] Building C object CMakeFiles/openapi_petstore.dir/model/user.c.o
[ 45%] Building C object CMakeFiles/openapi_petstore.dir/api/PetAPI.c.o
[ 50%] Building C object CMakeFiles/openapi_petstore.dir/api/StoreAPI.c.o
[ 54%] Building C object CMakeFiles/openapi_petstore.dir/api/UserAPI.c.o
[ 58%] Linking C shared library libopenapi_petstore.so
[ 58%] Built target openapi_petstore
Scanning dependencies of target unit-manual-PetAPI
[ 62%] Building C object CMakeFiles/unit-manual-PetAPI.dir/unit-tests/manual-PetAPI.c.o
[ 66%] Linking C executable unit-manual-PetAPI
[ 66%] Built target unit-manual-PetAPI
Scanning dependencies of target unit-manual-UserAPI
[ 70%] Building C object CMakeFiles/unit-manual-UserAPI.dir/unit-tests/manual-UserAPI.c.o
[ 75%] Linking C executable unit-manual-UserAPI
[ 75%] Built target unit-manual-UserAPI
Scanning dependencies of target unit-manual-StoreAPI
[ 79%] Building C object CMakeFiles/unit-manual-StoreAPI.dir/unit-tests/manual-StoreAPI.c.o
[ 83%] Linking C executable unit-manual-StoreAPI
[ 83%] Built target unit-manual-StoreAPI
Scanning dependencies of target unit-manual-user
[ 87%] Building C object CMakeFiles/unit-manual-user.dir/unit-tests/manual-user.c.o
[ 91%] Linking C executable unit-manual-user
[ 91%] Built target unit-manual-user
Scanning dependencies of target unit-manual-order
[ 95%] Building C object CMakeFiles/unit-manual-order.dir/unit-tests/manual-order.c.o
[100%] Linking C executable unit-manual-order
[100%] Built target unit-manual-order
Data is:{
	"id":	1234,
	"category":	{
		"id":	5,
		"name":	"Example Category"
	},
	"name":	"Example Pet",
	"photoUrls":	["http://www.github.com", "http://www.gitter.im"],
	"tags":	[{
			"id":	1,
			"name":	"beautiful code"
		}, {
			"id":	542353,
			"name":	"at least I tried"
		}],
	"status":	"available"
}
successful operation
Data is:{
	"id":	1234,
	"category":	{
		"id":	5,
		"name":	"Example Category"
	},
	"name":	"Rocky Handsome",
	"photoUrls":	["http://www.github.com", "http://www.gitter.im"],
	"tags":	[{
			"id":	1,
			"name":	"beautiful code"
		}, {
			"id":	542353,
			"name":	"at least I tried"
		}],
	"status":	"sold"
}
successful operation
User is: 
{
	"id":	1234,
	"username":	"example123",
	"firstName":	"Example1",
	"lastName":	"Example2Last",
	"email":	"example@example.com",
	"password":	"thisisexample!123",
	"phone":	"+123456789",
	"userStatus":	4
}
successful operation
Login User: logged in user session:1572762309163
successful operation
Placed order: 
{
	"id":	1234,
	"petId":	12345,
	"quantity":	50,
	"shipDate":	"2018-09-24T10:19:09.592+0000",
	"status":	"placed",
	"complete":	true
}
successful operation
Order received: 
{
	"id":	1234,
	"petId":	12345,
	"quantity":	50,
	"shipDate":	"2018-09-24T10:19:09.592+0000",
	"status":	"placed",
	"complete":	true
}
Order Deleted 
Order not found
successful operation
sold - 2
pending - 2
available - 9
```



Ref: https://travis-ci.org/OpenAPITools/openapi-generator/builds/606108932?utm_source=github_status&utm_medium=notification

Comment (michelealbano) [reply samples/client/petstore/c/CMakeLists.txt]: @wing328 I just run ./bin/c-petstore.sh after the changes by @zhemant 
Should we meet on Slack to decide how it should be?

File: modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache (hunk 1 above)
Comment (michelealbano) [reply modules/openapi-generator/src/main/resources/C-libcurl/model-body.mustache]: I didn't know it works for templates, I thought it got confused by the syntax.
Nice to know.

File: samples/client/petstore/c/CMakeLists.txt (hunk 2 above)
Comment (michelealbano) [reply samples/client/petstore/c/CMakeLists.txt]: @zhemant I gave a look at the new c-petstore.sh, and I haven't the expertise to correct the issue raised by @wing328 . Can you help with solving it?

Comment (zhemant) [reply samples/client/petstore/c/CMakeLists.txt]: @michelealbano  Sure,  I think I already have an idea why this is happening. If you can talk on IM, then I can maybe tell you in details about how the working is. I had sent you text on gitter. I don't know if you use it.

Comment (wing328) [reply samples/client/petstore/c/CMakeLists.txt]: I've filed https://github.com/OpenAPITools/openapi-generator/pull/4457 with some minor enhancements, which should correct the skipped tests due to regenerating the CMakefile file

Comment (wing328) [reply samples/client/petstore/c/CMakeLists.txt]: @michelealbano can you please merge the latest master into this branch as I've merged #4457 into master?

**Comment by wing328** [comment]
Successfully trigger build failure via https://travis-ci.org/OpenAPITools/openapi-generator/builds/617600938

**Comment by wing328** [comment]
@michelealbano thanks for the PR, which has been included in the v4.2.2 release: https://twitter.com/oas_generator/status/1201432648544972800

**Pull request merged**